File: server/src/main/java/by/brawl/entity/Squad.java
package by.brawl.entity;

import javax.persistence.*;
import java.util.LinkedHashSet;
import java.util.Set;

@Entity
public class Squad extends NamedEntity {

    @ManyToOne
    private Account owner;

    @ManyToMany
    @JoinTable(name = "hero_squad",
            joinColumns = @JoinColumn(name = "squad_id"),
            inverseJoinColumns= @JoinColumn(name = "hero_id"))
    private Set<Hero> heroes = new LinkedHashSet<>();

    public Account getOwner() {
        return owner;
    }

    public Set<Hero> getHeroes() {
        return heroes;
    }
}

File: server/src/main/java/by/brawl/entity/Squad.kt
package by.brawl.entity

import java.util.*
import javax.persistence.*

@Entity
class Squad : NamedEntity() {

    @ManyToOne
    lateinit var owner: Account

    @ManyToMany
    @JoinTable(name = "hero_squad",
            joinColumns = arrayOf(JoinColumn(name = "squad_id")),
            inverseJoinColumns = arrayOf(JoinColumn(name = "hero_id")))
    val heroes: Set<Hero> = LinkedHashSet()

}